tests/core/test_cluster_auth_tokens.py

```diff
@@ -0,0 +1,86 @@
+import subprocess
+from .conftest import create_kubeconfig, wait_for
+from kubernetes.client import CustomObjectsApi
+
+
+# test if the kubeconfig works to list api-resources for the fqdn context
+def exec_kubectl(dind_cc, headers, cmd='api-resources'):
+    cluster_kubeconfig_file = create_kubeconfig(dind_cc, headers)
+    # verify cluster scoped access
+    kubectl_cmd = 'kubectl ' + cmd + \
+                  ' --kubeconfig ' + cluster_kubeconfig_file + \
+                  ' --context ' + dind_cc.name + '-fqdn'
+    try:
+        return subprocess.check_output(kubectl_cmd, shell=True)
+    except subprocess.CalledProcessError as err:
+        print('kubectl error: ' + str(err.returncode))
+        raise err
+
+
+# test generator for multiple attempts
+def kubectl_available(dind_cc, headers):
+    def test():
+        try:
+            exec_kubectl(dind_cc, headers)
+            return True
+        except Exception:
```

Comment:
Specify the error type you are expecting here so other exceptions are not masked.